docs/content/en/docs/community/contributing.md

```diff
@@ -5,80 +5,129 @@ description: >
   How to best contribute to the project
 ---
 
-Thank you for your interest in contributing to the project. Whether it's
-a bug report, new feature, correction, or additional documentation, we
-greatly value feedback and contributions from the community.
+Thank you for your interest in contributing to our project. Whether it's a bug report, new feature, correction, or additional
+documentation, we greatly value feedback and contributions from our community.
 
-Please read through this document before submitting any issues or pull
-requests to ensure we have all the necessary information to respond 
-to your bug report or contribution effectively.
+Please read through this document before submitting any issues or pull requests to ensure we have all the necessary
+information to effectively respond to your bug report or contribution.
+
+## General Guidelines
+
+### Pull Requests
+
+Make sure to keep Pull Requests small and functional to make them easier to review, understand, and look up in commit history.
+This repository uses "Squash and Commit" to keep our history clean and make it easier to revert changes based on PR.
+
+Adding the appropriate documentation, unit tests and e2e tests as part of a feature is the responsibility of the
+feature owner, whether it is done in the same Pull Request or not.
+
+Pull Requests should follow the "subject: message" format, where the subject describes what part of the code is being
+modified.
+
+Refer to the [template](https://github.com/aws/eks-anywhere/blob/main/.github/PULL_REQUEST_TEMPLATE.md) for more information on what goes into a PR description.
+
+### Design Docs
+
+A contributor proposes a design with a PR on the repository to allow for revisions and discussions.
+If a design needs to be discussed before formulating a document for it, make use of GitHub Discussions to
+involve the community on the discussion.
+
+### GitHub Discussions
+
+GitHub Discussions are used for feature requests (that don't have actionable items/issues), questions, and anything else
+the community would like to share.
+
+Categories:
```

Comment:
Should we make some example discussions in the repo and link to them?

Reply:
I was planning on creating the Discussions structure in the meantime as well based on this review.